frontend/src/lib/utils.ts

```diff
@@ -203,3 +203,23 @@ export function labelVisibilityProtoValueToEnum(
       return LabelVisibilityOptions.Visible
   }
 }
+
+/**
+ * Tries to get an IFrame in which Streamlit app is embedded on Cloud deployments.
+ * It assumes iframe has title="streamlitApp" if it does not, it gets first found IFrame,
+ * otherwise returns null.
+ */
+export function getIFrameEnclosingApp(): HTMLIFrameElement | null {
+  if (!isInChildFrame()) {
+    return null
+  }
+  const qs = document.querySelectorAll('iframe[title="streamlitApp"]')
```

Comment:
It seems like it would be nice if we had a unique identifier that we place in our main iframe that we can use for this kind of thing.  Would that be hard to add?

Reply:
Ah, yeah doing something like this should work without being too difficult to implement, I think.

It'd be difficult to set an ID in the main iframe and use it here because the Cloud hosting service is what's embedding the Streamlit app in an iframe, and coordinating with the service to know what it set the iframe ID to is likely infeasible.

But I think we could give one of the `div`s in the app (in this case, the `StyledApp` styled-component) a randomly generated ID, then pass that ID to this helper function to use to verify that we've selected the right iframe. This would probably be worth trying, but I'm also okay with the current solution since I don't think the implications of weird behavior when using the print button with >1 Streamlit app embedded on a page will be much of a problem in practice